Feature/instalation guides xserver (#9)
ThundeRatz/ROSGazeboGuide: InstalationGuides/XServer.md

> ![Logo do XServer no windowsa](../assets/img/logo_Xserver.png)

Reviewer: ```suggestion
![Logo do XServer no windows](../assets/img/logo_Xserver.png)
```

# **Como Instalar o Xserver no Windows**

Neste tutorial, vamos aprender a **instalar** o Xorg Server chamado **Xserver** em sua máquina.

> 
> 

Reviewer: ```suggestion

```

Após a instalação do WSL e ROS, o seu sistema operacional ainda não consegue reproduzir o Gazebo e Rviz. Por conta disso, nesse tutorial iremos mostrar uma ferramenta que possibilita a a reprodução desses programas no seu Windows.

> Não é necessário ter o WSL e ROS/Gazebo instalado no seu computador para seguir esse tutorial, porém recomendo que façam esse tutorial por último.

Reviewer: ```suggestion
Não é necessário ter o WSL e ROS/Gazebo instalado no seu computador para seguir esse tutorial, porém é recomendado que se faça esse tutorial por último.
```

***[Clique aqui](../InstalationGuides/WSL.md)*** para ser direcionado ao tutorial de instalação do WSL 

***[Clique aqui](../InstalationGuides/ROSGazeboWSL.md)*** para ser direcionado ao tutorial de instalação do ROS/Gazebo

> 
> 

Reviewer: ```suggestion

```

## **Índice**<!-- omit in toc --> 

- [**Download e execução do arquivo VcXsrv**](#Download-e-execução-do-arquivo-VcXsrv)
- [**Execução e configuração do Xlaunch**](#Execução-e-configuração-do-Xlaunch)
- [**Instruções para pós instalação**](#Instruções-para-pós-instalação)

> 
> 

Reviewer: ```suggestion

```

## **Download e execução do arquivo VcXsrv**

> 
> 

Reviewer: ```suggestion

```

- **Baixe** o arquivo **VcXsrv Windows X Server** e **salve** no seu local de preferencia:

 ***[clique aqui](https://sourceforge.net/projects/vcxsrv/)*** para baixar o VcXsrv Windows X Server

> 
> 

Reviewer: ```suggestion

```

![download XServer](../assets/gif/XServer/0_baixando_VcXsrv.gif)

> 
> 

Reviewer: ```suggestion

```

> - Execute e instale o arquivo **"vcxsrv-64.1.20.9.0.installer"**

Reviewer: Aqui acho melhor sei lá colocar uns "x" no lugar dos números e falar que eles estão no lugar da versão, porque vai que atualizam a versão, não precisa ser essa daí em específico.

    
    - O arquivo, a priori, já vem com as configurações adequadas, porém, certifique-se que a opção de instalação está "full" e que todas as "caixinhas" estão selecionadas. 

> 
> 

Reviewer: ```suggestion

```

![execute o XServer](../assets/gif/XServer/1_executando_VcXsrv.gif)

> 
> 

Reviewer: ```suggestion

```

## **Execução e configuração do Xlaunch**

> Após baixar e executar o arquivo "vcxsrv-64.1.20.9.0.installer", ele irá criar um arquivo chamado **Xlaunch** na sua área de trabalho (Desktop, se o seu computador estiver em inglês)

Reviewer: Idem sobre a versão aqui. ~~Mas nossa deixar na área de trabalho hasuahuahushaush~~

- Execute o **Xlaunch**
- Na aba de configuração de display (select display settings):
    - Deixe **"Multiple windows"** selecionado;

>     - Deixe **display number = 0**.

Reviewer: ```suggestion
    - Deixe **"display number"** em **0**.
```

- Na aba de seleção de como iniciar o cliente (select how to start clients):
    - Selecione **"Start no client"**.

- Na aba de configurações extras (extra settings):
    - Deixe selecionado **"Clipboard"**, **"Primary Selection"** e **"Disable acess control"**

> 
> 

Reviewer: ```suggestion

```

![executando Xlaunch](../assets/gif/XServer/2_executando_Xlaunch.gif)

> 
> 
> 

Reviewer: ```suggestion

```

## **Instruções para pós instalação**

> 
> 

Reviewer: ```suggestion

```

<br>

- Execute e instale o arquivo **"vcxsrv-64.1.20.9.0.installer"**
    
    - O arquivo, em priori, já vem com as configurações adequadas, porém, certifique-se que a opção de instalação está "full" e todas as "caixinhas" selecionadas. 

<br>

![execute o XServer](../assets/gif/XServer/1_executando_VcXsrv.gif)

<br>

## **Execução e configuração do Xlaunch**

<br>

Após baixar e executar o arquivo "vcxsrv-64.1.20.9.0.installer", ele irá criar um arquivo chamado **Xlaunch** na sua área de trabalho (Desktop, se o seu computador estiver em inglês)

- Execute o **Xlaunch**
- Na aba de configuração de display(select display settings):
    - Deixe **"Multiple windows"** selecionado;
    - Deixe **display number = 0**.

- Na aba de seleção de como iniciar o cliente(select how to start clients):
    - Selecione **"Start no client"**.

- Na aba de configurações extras(extra settings):
    - Deixe selecionado **"Clipboard"**, **"Primary Selection"** e **"Disable acess control"**

<br>

![executando Xlaunch](../assets/gif/XServer/2_executando_Xlaunch.gif)

<br>

## **Instruções para pós instalação**

<br>

Depois disso, está pronta a instalação do Xserver.

> Após a instalação do WSL, ROS/Gazebo e Xserver, o seu computador está pronto para as experiencias que serão realizadas nas aulas!

Reviewer 2: ```suggestion
Após a instalação do WSL, ROS/Gazebo e Xserver, o seu computador está pronto para as experiências que serão realizadas nas aulas!
```

Bom proveito!

>  ***[clique aqui]()***  para executar todo o projeto no seu Windows!

Reviewer 2: Sla, parece que clicando aí ele vai executar ja tudo para você auhsuahshuhashhu

```suggestion
Para aprender a executar seu projeto no Windows, ***[clique aqui]()*** 
```

> parent: Guias de Instalação

Reviewer: Tem lugar que tá "x server" e lugar que tá "xserver", acho que dá pra padronizar com "x server". ~~Eu comentei nessa linha, mas é na de cima que tem q mexer, eh que não tinha como comentar lá.~~ 

Tem que arrumar no ìndice principal do repositório também e no arquivo de instalação do ROS/Gazebo no WSL.

nav_order: 4
---

![Logo do XServer no Windows](../assets/img/logo_Xserver.png)

> # **Como Instalar o Xserver no Windows**

Reviewer: ```suggestion
# **Como Instalar o X server no Windows**
```

> Neste tutorial, vamos aprender a **instalar** o Xorg Server chamado **Xserver** em sua máquina.

Reviewer: ```suggestion
Neste tutorial, vamos aprender a **instalar** o Xorg Server chamado **X server** em sua máquina.
```

Após a instalação do WSL e ROS, o seu sistema operacional ainda não consegue reproduzir o Gazebo. Por conta disso, nesse tutorial iremos mostrar uma ferramenta que possibilita a reprodução desses programas no Windows.

> Não é necessário ter o ROS/Gazebo instalado no seu computador para seguir esse tutorial, mas é necessario o WSL para os toques finais. Contudo, recomendo que faça esse tutorial por último.

Reviewer: ```suggestion
Não é necessário ter o ROS/Gazebo instalado no seu computador para seguir esse tutorial, mas é necessario o WSL para os toques finais. Dessa forma, recomendo que faça esse tutorial por último.
```

***[Clique aqui](../InstalationGuides/WSL.md)*** para ser direcionado ao tutorial de instalação do WSL.

***[Clique aqui](../InstalationGuides/ROSGazeboWSL.md)*** para ser direcionado ao tutorial de instalação do ROS/Gazebo.
## **Índice**<!-- omit in toc --> 

- [**Download e execução do arquivo VcXsrv**](#Download-e-execução-do-arquivo-VcXsrv)
- [**Execução e configuração do Xlaunch**](#Execução-e-configuração-do-Xlaunch)
- [**Toques finais**](#Toques-finais)
- [**Teste de funcionamento**](#Teste-de-funcionamento)
- [**Instruções para pós instalação**](#Instruções-para-pós-instalação)

## **Download e execução do arquivo VcXsrv**

- **Baixe** o arquivo **VcXsrv Windows X Server** e **salve** no seu local de preferência:

    - ***[Clique aqui](https://sourceforge.net/projects/vcxsrv/)*** para baixar o VcXsrv Windows X Server.

![download XServer](../assets/gif/XServer/0_baixando_VcXsrv.gif)

> - Execute e instale o arquivo **"vcxsrv-64.1.x.x.x.installer"** (obs.: os 'x' estão no lugar da versão, ou seja, desde que estaja dentro da "vcxsrv-64.1", está tudo bem baixar essa opção)

Reviewer: ```suggestion
- Execute e instale o arquivo **"vcxsrv-64.1.x.x.x.installer"** (obs.: os 'x' estão no lugar da versão, ou seja, desde que estaja dentro da "vcxsrv-64.1", está tudo bem baixar essa opção).
```

    
    - O arquivo, a priori, já vem com as configurações adequadas, porém, certifique-se que a opção de instalação está "full" e que todas as "caixinhas" estão selecionadas. 
    
![execute o XServer](../assets/gif/XServer/1_executando_VcXsrv.gif)

## **Execução e configuração do Xlaunch**

Após baixar e executar o arquivo "vcxsrv-64.1.x.x.x.installer", ele irá criar um arquivo chamado **Xlaunch** na sua área de trabalho (desktop, se o seu computador estiver em inglês).

Toda vez que computador for reiniciado, esse passo deve ser feito novamente.

> - Execute o **Xlaunch**

Reviewer: ```suggestion
- Execute o **Xlaunch**.
```

- Na aba de configuração de display (select display settings):
    - Deixe **"Multiple windows"** selecionado;
    - Deixe **"display number"** em **0**.

- Na aba de seleção de como iniciar o cliente (select how to start clients):
    - Selecione **"Start no client"**.

- Na aba de configurações extras (extra settings):
    - Deixe selecionado **"Clipboard"**, **"Primary Selection"** e **"Disable acess control"**.

Ao final da instalação, há a opção de salvar as configurações feita, para que não precise refazer os mesmos passos acima toda vez que for iniciar o X Server.

- Salvar configurações(opcinal):
    - **Clique** no icone **"Save Configuration"**;

>     - **Nomeie** o arquivo como preferir e **Salve** no seu local de preferência.

Reviewer: ```suggestion
    - **Nomeie** o arquivo como preferir e **salve** no seu local de preferência.
```

Depois disso, basta clicar no arquivo de configurações salvo para iniciar o X Server.

![executando Xlaunch](../assets/gif/XServer/2_executando_Xlaunch.gif)

- Para saber se foi inicializado, certifique-se que existe na extremidade inferior direita o símbolo do Xlaunch.

![Verificação Xlaunch](../assets/img/check_xlaunch.png)

## **Toques finais**

Para conseguir fazer esses toques finais, é necessário ter, pelo menos, o WSL instalado no Windows.

> Caso não tenha feito isso ainda, ***[Clique aqui](../InstalationGuides/WSL.md)*** para ser direcionado ao tutorial de instalação do WSL.

Reviewer: ```suggestion
Caso não tenha feito isso ainda, ***[clique aqui](../InstalationGuides/WSL.md)*** para ser direcionado ao tutorial de instalação do WSL.
```

> - **Abra** o **terminal** do Ubuntu no Windows e digite o seguinte comando:

Reviewer: ```suggestion
- Caso esteja usando o **WSL1**, **abra** o **terminal** do Ubuntu no Windows e digite o seguinte comando:
```

```bash
echo "export DISPLAY=:0" >> ~/.bashrc
source ~/.bashrc
```
Não se preocupe se o terminal não responder nada depois destes comandos -- ele realmente não dá nenhum sinal de que os comandos foram feitos de maneira correta. Na verdade, se não apareceu nada, você provavelmente fez tudo certo.

> Observação:

Reviewer: ```suggestion
```

- Caso esteja usando o WSL2, digite o seguinte no lugar do último comando:

```bash
echo "export DISPLAY=$(cat /etc/resolv.conf | grep nameserver | awk '{print $2}'):0" >> ~/.bashrc
source ~/.bashrc
```

## **Teste de funcionamento**
Com todos os passos acima concluidos, você só precisa certificar-se que está tudo funcionando.

- **Abra** o **terminal** do Ubuntu e **Instale** o pacote `mesa-utils` com o seguinte comando:

```bash
sudo apt install mesa-utils
```

![mesa-utils instalation](../assets/gif/XServer/3_mesa_utils_instalation.gif)

- **Digite** no terminal o seguinte comando:

```bash
glxgears
```

Depois desse comando, deve aparecer três engrenagens coloridas girando:

![gears](../assets/gif/XServer/4_gears.gif)

## **Instruções para pós instalação**

> Depois disso, está pronta a instalação do Xserver.
> 
> Após a instalação do WSL, ROS/Gazebo e Xserver, o seu computador está pronto para as experiências que serão realizadas nas aulas!

Reviewer: ```suggestion
Depois disso, está pronta a instalação do X server.

Após a instalação do WSL, ROS/Gazebo e X server, o seu computador está pronto para as experiências que serão realizadas nas aulas!
```

Bom proveito!

> Para aprender a executar seu projeto no Windows, ***[clique aqui](../HowToRun/RunOnWSL.md)*** 

Reviewer: ```suggestion
Para aprender a executar seu projeto no Windows, ***[clique aqui](../HowToRun/RunOnWSL.md)***. 
```

***[Clique aqui](../InstalationGuides/ROSGazeboWSL.md)*** para ser direcionado ao tutorial de instalação do ROS/Gazebo.
## **Índice**<!-- omit in toc --> 

- [**Download e execução do arquivo VcXsrv**](#Download-e-execução-do-arquivo-VcXsrv)
- [**Execução e configuração do Xlaunch**](#Execução-e-configuração-do-Xlaunch)
- [**Instruções para pós instalação**](#Instruções-para-pós-instalação)

## **Download e execução do arquivo VcXsrv**

- **Baixe** o arquivo **VcXsrv Windows X Server** e **salve** no seu local de preferencia:

    - ***[Clique aqui](https://sourceforge.net/projects/vcxsrv/)*** para baixar o VcXsrv Windows X Server.

![download XServer](../assets/gif/XServer/0_baixando_VcXsrv.gif)

- Execute e instale o arquivo **"vcxsrv-64.1.x.x.x.installer"** (obs.: os 'x' estão no lugar da versão, ou seja, desde que estaja dentro da "vcxsrv-64.1", está tudo bem baixar essa opção)
    
    - O arquivo, a priori, já vem com as configurações adequadas, porém, certifique-se que a opção de instalação está "full" e que todas as "caixinhas" estão selecionadas. 
    
![execute o XServer](../assets/gif/XServer/1_executando_VcXsrv.gif)

## **Execução e configuração do Xlaunch**

> Após baixar e executar o arquivo "vcxsrv-64.1.x.x.x.installer", ele irá criar um arquivo chamado **Xlaunch** na sua área de trabalho (Desktop, se o seu computador estiver em inglês)

Reviewer 3: ```suggestion
Após baixar e executar o arquivo "vcxsrv-64.1.x.x.x.installer", ele irá criar um arquivo chamado **Xlaunch** na sua área de trabalho (desktop, se o seu computador estiver em inglês)
```
Se tá com letra minúscula em portuguẽs, coloca em letra minúscula em inglês também

Toda vez que computador for reiniciado, esse passo deve ser feito novamente.

- Execute o **Xlaunch**
- Na aba de configuração de display (select display settings):
    - Deixe **"Multiple windows"** selecionado;
    - Deixe **"display number"** em **-1**.

- Na aba de seleção de como iniciar o cliente (select how to start clients):
    - Selecione **"Start no client"**.

- Na aba de configurações extras (extra settings):
    - Deixe selecionado **"Clipboard"**, **"Primary Selection"** e **"Disable acess control"**

![executando Xlaunch](../assets/gif/XServer/2_executando_Xlaunch.gif)

- Para saber se está funcionando, certifique-se que existe na extremidade inferior direita o símbolo do Xlaunch.

![Verificação Xlaunch](../assets/img/check_xlaunch.png)

## **Teste de funcionamento**

Para conseguir fazer o teste de funcionamento, é necessário ter, pelo menos, o WSL instalado no Windows.

Caso não tenha feito isso ainda, volte para o inicio da página  [**clicando aqui**](#Como-Instalar-o-Xserver-no-Windows) e siga o tutorial de instalação do WSL.

- **Abra** o **terminal** do Ubuntu no Windows
- **Abra** o arquivo ".bashrc"

Para abrir esse arquivo digite seguinte comando:
```bash

> nano ~/.bashrc

Reviewer 3: No [guia de instalação do ROS](https://github.com/ThundeRatz/ROSGazeboGuide/blob/main/InstalationGuides/ROSGazeboWSL.md#toques-finais), nós fazemos uma edição no `.bashrc` direto pela linha de comando, acredito que podemos fazer o mesmo aqui pra simplificar o processo

Ao invés de 

```bash
nano ~/.bashrc
export DISPLAY=:0 # linha adicionada no arquivo
```

Podemos fazer direto

```bash
echo "export DISPLAY=:0" >> ~/.bashrc
```

O caminho tem menos passos intermediários. Outra mudança que eu acredito que é interessante é mover essas instruções pra cima, pois elas fazem parte do conjunto "Instruções de instalação", não "Instruções de teste". Digo, não é necessário fazer isso _toda vez_ que se abrir o XServer, só no momento da instação

```

- **Digite** o seguinte comando no final do arquivo:

```bash
export DISPLAY=:0
```

Observação:
- Caso esteja usando o WSL2, digite o seguinte no final do arquivo no lugar do último comando:

```bash
export DISPLAY=$(cat /etc/resolv.conf | grep nameserver | awk '{print $2}'):0
```

- **Aperte** "Ctrl+O" para salvar as mudanças feitas
- **Aperte** "Ctrl+X" para sair do arquivo

![executando Xlaunch](../assets/gif/XServer/3_display_bashrc.gif)

- **Digite** o seguintes comando:

```bash
source ~/.bashrc
```

Não se preocupe se o terminal não responder nada depois deste comando -- ele realmente não dá nenhum sinal de que o comando foi feito de maneira correta. Na verdade, se não apareceu nada, você provavelmente fez tudo certo.

> - **Instale** o pacote `mesa-utilis` com seguinte comando:

Reviewer 3: ```suggestion
- **Instale** o pacote `mesa-utilis` com o seguinte comando:
```

```bash
sudo apt install mesa-utils
```

![mesa-utils instalation](../assets/gif/XServer/4_mesa_utils_instalation.gif)

> - **Digite** no terminal o seguintes comando:

Reviewer 3: ```suggestion
- **Digite** no terminal o seguinte comando:
```

```bash
glxgears
```

> Depois dessa série de comando, deve aparecer três engrenagens coloridas girando:

Reviewer 3: ```suggestion
Depois dessa série de comandos, deve aparecer três engrenagens coloridas girando:
```

parent: Guias de Instalação
nav_order: 4
---
![Logo do XServer no Windows](../assets/img/logo_Xserver.png)

# **Como Instalar o Xserver no Windows**

Neste tutorial, vamos aprender a **instalar** o Xorg Server chamado **Xserver** em sua máquina.
Após a instalação do WSL e ROS, o seu sistema operacional ainda não consegue reproduzir o Gazebo. Por conta disso, nesse tutorial iremos mostrar uma ferramenta que possibilita a reprodução desses programas no Windows.

Não é necessário ter o WSL e ROS/Gazebo instalado no seu computador para seguir esse tutorial, porém é recomendado que se faça esse tutorial por último.

> ***[Clique aqui](../InstalationGuides/WSL.md)*** para ser direcionado ao tutorial de instalação do WSL 
> 
> ***[Clique aqui](../InstalationGuides/ROSGazeboWSL.md)*** para ser direcionado ao tutorial de instalação do ROS/Gazebo

Reviewer: ```suggestion
***[Clique aqui](../InstalationGuides/WSL.md)*** para ser direcionado ao tutorial de instalação do WSL.

***[Clique aqui](../InstalationGuides/ROSGazeboWSL.md)*** para ser direcionado ao tutorial de instalação do ROS/Gazebo.
```

## **Índice**<!-- omit in toc --> 

- [**Download e execução do arquivo VcXsrv**](#Download-e-execução-do-arquivo-VcXsrv)
- [**Execução e configuração do Xlaunch**](#Execução-e-configuração-do-Xlaunch)
- [**Instruções para pós instalação**](#Instruções-para-pós-instalação)

> ## **Download e execução do arquivo VcXsrv**
> - **Baixe** o arquivo **VcXsrv Windows X Server** e **salve** no seu local de preferencia:

Reviewer: ```suggestion

## **Download e execução do arquivo VcXsrv**

- **Baixe** o arquivo **VcXsrv Windows X Server** e **salve** no seu local de preferencia:
```

>  ***[clique aqui](https://sourceforge.net/projects/vcxsrv/)*** para baixar o VcXsrv Windows X Server

Reviewer: ```suggestion
    - ***[Clique aqui](https://sourceforge.net/projects/vcxsrv/)*** para baixar o VcXsrv Windows X Server.
```

> ![download XServer](../assets/gif/XServer/0_baixando_VcXsrv.gif)
> - Execute e instale o arquivo **"vcxsrv-64.1.x.x.x.installer"** (obs.: os 'x' estão no lugar da versão, ou seja, desde que estaja dentro da "vcxsrv-64.1", está tudo bem baixar essa opção)
>     
>     - O arquivo, a priori, já vem com as configurações adequadas, porém, certifique-se que a opção de instalação está "full" e que todas as "caixinhas" estão selecionadas. 
> ![execute o XServer](../assets/gif/XServer/1_executando_VcXsrv.gif)

Reviewer: ```suggestion

![download XServer](../assets/gif/XServer/0_baixando_VcXsrv.gif)

- Execute e instale o arquivo **"vcxsrv-64.1.x.x.x.installer"** (obs.: os 'x' estão no lugar da versão, ou seja, desde que estaja dentro da "vcxsrv-64.1", está tudo bem baixar essa opção)
    
    - O arquivo, a priori, já vem com as configurações adequadas, porém, certifique-se que a opção de instalação está "full" e que todas as "caixinhas" estão selecionadas. 
    
![execute o XServer](../assets/gif/XServer/1_executando_VcXsrv.gif)

```

Author: acredite ou não, mas eu tinha deixado tudo juntinha pq achei que vc iria preferir hahahah

## **Execução e configuração do Xlaunch**

Após baixar e executar o arquivo "vcxsrv-64.1.x.x.x.installer", ele irá criar um arquivo chamado **Xlaunch** na sua área de trabalho (Desktop, se o seu computador estiver em inglês)
Toda vez que computador for reiniciado, esse passo deve ser feito novamente.

- Execute o **Xlaunch**
- Na aba de configuração de display (select display settings):
    - Deixe **"Multiple windows"** selecionado;

>     - Deixe **"display number"** em **-1**.

Author: vou ter que mudar o gif de novo hahaha mas ok

>     - Deixe **"display number"** em **-1**.

Reviewer: Noooosssa justo verdade, sorry shuahsuahsuahsu. Mas ah só reverter esse commit aqui 01633ba0bd97c906cc36ad66957f8ded766f18b6 né
